azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -121,122 +128,107 @@ private FlowContainer(final Path projectDirPath, final Props props)
+   * execution. It is assumed that AZ_HOME environment variable is set. If it is not set, then
+   * it explicitely sets it to present working directory.
+   * @param args Takes the execution id and Project zip file path as inputs.
+   * @throws IOException
+   * @throws ExecutorManagerException
+   */
   public static void main(final String[] args) throws IOException, ExecutorManagerException {
-    // Redirect all std out and err messages into log4j
+    // Redirect all std out and err messages into slf4j
     StdOutErrRedirect.redirectOutAndErrToLog();
     // Get all the arguments
-    final String execIdStr = args[0];
-    final String projectZipName = args[1];
+    final String execIdStr = args[ARG_EXEC_ID];
+    final String projectZipName = args[ARG_PROJECT_LOCATION];
     // Process Execution ID.
-    int execId = 0;
+    int execId = -1;
     try {
       execId = Integer.parseInt(execIdStr);
     } catch (NumberFormatException ne) {
-      System.out.printf("Execution ID %s is invalid %n", args[0]);
+      logger.error("Execution ID passed in argument is invalid {}", execIdStr);
+      throw new ExecutorManagerException(ne);
     }
-    System.out.printf("Execution ID = %d%n", execId);
 
-    // Setup work directories
-    final Path currentWorkingDir = Paths.get("").toAbsolutePath();
-    final Path projectDirPath = Paths.get(currentWorkingDir.toString(), PROJECT_DIR);
-    final Path jobtypePluginPath = Paths.get(currentWorkingDir.toString(), JOBTYPE_DIR);
+    logger.info("Execution ID : " + execId);
 
-    // Move files to respective dirs
-    // Create project dir
-    // TODO: Need to revisit
-    System.out.println("Creating project dir");
-    Files.createDirectory(projectDirPath);
-    Path projectZipPath = Paths.get(projectDirPath.toString(), projectZipName);
-    System.out.println("moving projectDir:" + projectZipName + ": to " +
-        projectZipPath);
-    Files.move(Paths.get(projectZipName), projectZipPath);
-    // TODO : revisit this logic,
-    // Unzip the project zip
-    FlowContainer.unzipFile(projectZipPath.toString(), projectDirPath);
+    // AZ_HOME must provide a correct path, if not then azHome is set to current working dir.
+    final String azHome = Optional.ofNullable(System.getenv(AZ_HOME)).orElse("");
 
-    // TODO : Deepak - throw away after directory setup is final.
-    Path pwd = Paths.get("").toAbsolutePath();
-    System.out.println("Deepak : Dumping dir structure for = " + pwd.toString());
-    java.nio.file.Files.walk(pwd).filter(java.nio.file.Files::isRegularFile)
-        .forEach(System.out::println);
+    final Path currentDir = Paths.get(System.getenv(AZ_HOME)).toAbsolutePath();
 
     // Set Azkaban props
-    Props props = setAzkabanProps(jobtypePluginPath);
+    final Path jobtypePluginPath = Paths.get(currentDir.toString(), JOBTYPE_DIR);
+    Props azkabanProps = FlowContainer.setAzkabanProps(jobtypePluginPath);
```

Comment:
No need to use FlowContainer.setAzkabanProps. You can directly use setAzkabanProps.